operators/pkg/instautoctrl/README.md

```diff
@@ -0,0 +1,93 @@
+# Instance Automation Controller
+
+The **Instance Automation Controller** (`instautoctrl` package) is part of the automation framework developed for the *Cloud Programming* course, Politecnico di Torino, 2025 edition.  
+It implements backend logic to manage the lifecycle of instances declared as **Inactive**.
+
+The package includes four controllers:
+
+- [Instance Inactive Termination](#instance-inactive-termination-controller)
+- [Instance Termination](#instance-termination-controller)
+- [Instance Submission](#instance-submission-controller)
+- [Instance Expiration](#instance-expiration-controller)
+
+
+## Instance Inactive Termination Controller
+This controller monitors instances and automates actions based on their inactivity status and lifespan. The controller understands if the Instance can be declared as Inactive and starts sending notifications to its tenant to inform them to access their Instance resources, otherwise they will be paused (if persistent) or deleted (if not persistent) after a specific period of time defined in the `Template` resource.
+
+
+### Detailed Behavior
+The controller begins by retrieving all the active **Instances**. For each instance, it determines whether it should be monitored or not. This decision is influenced by a special label that can be added to the associated **Namespace** resource, called `InstanceInactivityIgnoreNamespace`. If the Namespace carries this label, the controller will ignore the instance, allowing it to remain inactive for an extended period without being stopped or deleted.
+
+Once it is established that the instance should be monitored, the controller adds several annotations to it. These include the `AlertAnnotationNum`, which tracks the number of notifications sent to inform the tenant that the instance has been idle for some time and will soon be stopped or deleted. This number ranges from zero up to a maximum limit defined by `InstanceMaxNumberOfAlerts`, a custom parameter defined via Helm chart. This value could be overwritten by the `CustomNumberOfAlertsAnnotation` annotation in the associated Template resource. Another annotation is the `LastActivityAnnotation`, which records the last time the user accessed the instance either via the frontend through the Ingress or via SSH (info available through the SSH bastion tracker). The controller also adds the `LastNotificationTimestampAnnotation`, indicating the timestamp when the last notification was sent (if enabled). This timestamp helps to determine if enough time has passed since the previous notification, allowing a new alert to be sent if needed.
```

Comment:
Please specify how the time interval between different notifications is determined.